File: pb_migrations/1751800000_create_players.js
/// <reference path="../pb_data/types.d.ts" />
migrate((app) => {
  const collection = new Collection({
    name: "players",
    type: "base",
    fields: [
      {
        name: "oidc_sub",
        type: "text",
        required: true,
        min: 1,
        max: 200,
      },
      {
        name: "entity_id",
        type: "text",
        required: true,
        min: 1,
        max: 100,
      },
      {
        name: "display_name",
        type: "text",
        required: false,
        max: 100,
      },
      {
        name: "pos_x",
        type: "number",
        required: false,
      },
      {
        name: "pos_y",
        type: "number",
        required: false,
      },
    ],
    listRule: "",
    viewRule: "",
    createRule: null,
    updateRule: null,
    deleteRule: null,
  });

  return app.save(collection);
}, (app) => {
  const collection = app.findCollectionByNameOrId("players");
  return app.delete(collection);
});

File: backend/migrations/1751800000_create_players.go
package migrations

import (
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/tools/types"
	m "github.com/pocketbase/pocketbase/migrations"
)

func init() {
	m.Register(func(app core.App) error {
		collection := core.NewBaseCollection("players")

		collection.Fields.Add(
			&core.TextField{
				Name:     "oidc_sub",
				Required: true,
				Min:      1,
				Max:      200,
			},
			&core.TextField{
				Name:     "entity_id",
				Required: true,
				Min:      1,
				Max:      100,
			},
			&core.TextField{
				Name:     "display_name",
				Required: false,
				Max:      100,
			},
			&core.NumberField{
				Name:     "pos_x",
				Required: false,
			},
			&core.NumberField{
				Name:     "pos_y",
				Required: false,
			},
		)

		collection.ListRule = types.Pointer("")
		collection.ViewRule = types.Pointer("")
		collection.CreateRule = types.Pointer("")
		collection.UpdateRule = types.Pointer("")
		collection.DeleteRule = types.Pointer("")

		return app.Save(collection)
	}, func(app core.App) error {
		collection, err := app.FindCollectionByNameOrId("players")
		if err != nil {
			return err
		}
		return app.Delete(collection)
	})
}
